### ISSUE TITLE
Add SFT evals

### ISSUE THREAD

**Issue by ruimalheiro** [ruimalheiro/gradient-garden#93]
Currently in the SFT path we just run the standard validation loss. We do not run automatic evals that measure if the model is improving when it comes to instruction following during training which makes it harder to understand the progress and to compare with other checkpoints.

Expected work:
- Add support for a standard eval (good candidate is IFEval);
- Potentially add custom eval (that aligns with our synthetic data targets). Ideally score by category. Categories could be:
  - identity
  - stop/EOS behavior
  - exact yes/no answers
  - exact one word answers
  - exact N comma-separated items
  - exact N numbered lists
  - grammar correction
  - rewrite/clarity
  - summarization
  - friendly reply
  - simple factual QA
  - simple math
  - sentiment classification
- Reuse WinoGrande, HellaSwag and ARC-Challenge to support the evals if possible. Note: These are currently only used in pretraining, for sft we need to take in consideration the prompt structure / response as it will not be simple input completion. Also they are just regression checks not the main SFT metric to aim for obviously.
